plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/VerifierErrorInterpreter.kt

```diff
@@ -37,6 +39,12 @@ class VerifierErrorInterpreter {
                 reportOn(source, PluginErrors.UNEXPECTED_RETURNED_VALUE, "null", context)
             error is PostconditionViolated && role is SourceRole.CallsInPlaceEffect ->
                 reportOn(source, PluginErrors.INVALID_INVOCATION_TYPE, role.paramSymbol, role.kind.asUserFriendlyMessage, context)
+            (error is PreconditionInCallFalse || error is AssertFailed) && role is SourceRole.ParamFunctionLeakageCheck -> {
```

Comment:
The `error is ...` checks add a lot of noise to the code.  Is there any value to them?  Would our behaviour be different if the error came from something else?

Reply:
With the current errors we are verifying no, but I think in the future it can be relevant.

Reply:
Could you give an example?

Reply:
Mhh, now I think it off better, you were proposing to remove the `error is ...` checks at all, and then just rely on the type of `role`, right? If that is the case, then yes, the checks are redundant. At the moment, I cannot think of an example where the checks matter.

Reply:
I am going to refactor this part of the code in another PR I will mention here. The PR will remove new unused code.